File: app/src/main/java/de/westnordost/streetcomplete/quests/validator/DetectHistoricRailwayTagging.java
package de.westnordost.streetcomplete.quests.validator;

import android.os.Bundle;

import java.util.Map;

import javax.inject.Inject;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;


public class DetectHistoricRailwayTagging extends SimpleOverpassQuestType {
	@Inject
	public DetectHistoricRailwayTagging(OverpassMapDataDao overpassServer) { super(overpassServer); }

	@Override protected String getTagFilters()
	{
		return "nodes, ways, relations with railway=abandoned";
	}

	public AbstractQuestAnswerFragment createForm()
	{
		return new YesNoQuestAnswerFragment();
	}

	public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
	{}


	@Override public String getCommitMessage() { return null; }
	@Override public int getIcon() { return R.drawable.ic_quest_railway; }
	@Override public int getTitle(Map<String,String> tags)
	{
		return R.string.quest_railway_abandoned_detected;
	}
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/validator/DetectHistoricRailwayTagging.kt
package de.westnordost.streetcomplete.quests.validator

import de.westnordost.streetcomplete.R
import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement
import de.westnordost.streetcomplete.osm.Tags
import de.westnordost.streetcomplete.quests.YesNoQuestForm

class DetectHistoricRailwayTagging() : OsmFilterQuestType<Boolean>() {

    override val elementFilter = """nodes, ways, relations with railway~abandoned|dismantled|historic|razed
        or abandoned:railway
        or razed:railway
        or was:railway
        or removed:railway
        or historic:railway
        or demolished:railway""".trimMargin()
    override val changesetComment = ""
    override val icon = R.drawable.ic_quest_railway

    override fun getTitle(tags: Map<String, String>) = R.string.quest_railway_abandoned_detected

    override fun createForm() = YesNoQuestForm()

    override fun applyAnswerTo(answer: Boolean, tags: Tags, timestampEdited: Long) {
    }

    override val wikiLink = "Tag:railway=abandoned"

    override val achievements: List<EditTypeAchievement> = listOf()
}
